x-pack/plugins/fleet/server/services/preconfiguration.ts

```diff
@@ -438,3 +482,87 @@ async function addPreconfiguredPolicyPackages(
     );
   }
 }
+
+interface BundledPackageInstallResult extends InstallResult {
+  name: string;
+}
+
+/**
+ * Pulls bundled .zip archives of package from a directory and installs the packages. This facilitates
+ * "stack-aligned" packages that are bundled with a given release of Kibana.
+ */
+async function installBundledPackages(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  spaceId: string
+): Promise<BundledPackageInstallResult[]> {
+  const logger = appContextService.getLogger();
+
+  const bundledPackages = await getBundledPackages();
+
+  const results = await Promise.allSettled(
+    bundledPackages.map(async (bundledPackage) => {
+      // TODO: Determine a better way to simply grab the package name/version from the archive
+      const { packageInfo } = await parseAndVerifyArchiveEntries(
+        bundledPackage.buffer,
+        'application/zip'
+      );
+
+      const installedPkg = await getInstallationObject({
+        savedObjectsClient: soClient,
+        pkgName: packageInfo.name,
+      });
+
+      const installType = getInstallType({ pkgVersion: packageInfo.version, installedPkg });
+
+      // If the package is already installed, don't attempt to reinstall it. Just add a result record indicating
+      // that the package is already installed.
+      if (installType !== 'install') {
+        return {
+          name: installedPkg?.attributes.name,
+          status: 'already_installed',
+          installType: 'reinstall',
+        } as BundledPackageInstallResult;
+      }
+
+      logger.debug(`Installing bundled package ${bundledPackage.name}`);
+
+      const installResult = await installPackage({
+        savedObjectsClient: soClient,
+        esClient,
+        installSource: 'upload',
+        archiveBuffer: bundledPackage.buffer,
+        contentType: 'application/zip',
+        spaceId,
+      });
+
+      // Mark bundled packages with a unique status to prevent them from appearing as "installed" in the UI
+      await updateInstallStatus({
+        savedObjectsClient: soClient,
+        pkgName: packageInfo.name,
+        status: 'installed_bundled',
+      });
+
+      return {
+        // Strip semver strings from the package file name to make them consistent with the `name`
+        // field that comes through via preconfigured packages/policies
+        name: bundledPackage.name.substring(0, bundledPackage.name.indexOf('-')),
```

Comment:
Do we enforce package name to not contains `-`?

Reply:
Looks like yes, we only allow `a-z`, `0-9`, and `_`. 

https://github.com/elastic/package-spec/blob/31cf17fa121c5aed85f4506e970ea89ea8a0be26/versions/1/manifest.spec.yml#L115